compose/automation/expr_types.go

```diff
@@ -518,3 +518,104 @@ func CastToAttachment(val interface{}) (out *types.Attachment, err error) {
 		return nil, fmt.Errorf("unable to cast type %T to %T", val, out)
 	}
 }
+
+func EmptyComposeRecordValues() *ComposeRecordValues {
+	return &ComposeRecordValues{value: &types.Record{Values: types.RecordValueSet{}}}
+}
+
+func (t *ComposeRecordValues) Each(fn func(k string, v expr.TypedValue) error) (err error) {
+	if t.IsEmpty() {
+		return
+	}
+
+	for _, vv := range t.GetValue() {
+		key := vv.Name
+		if len(key) == 0 {
+			continue
+		}
+
+		var val expr.TypedValue
+		val, err = expr.Typify(vv)
+		if err = fn(key, val); err != nil {
+			return
+		}
+	}
+
+	return
+}
+
+// Merge combines the given ComposeRecordValues into ComposeRecordValues
+// NOTE: It will return CLONE of the original ComposeRecordValues, if it's called without any parameters
+func (t *ComposeRecordValues) Merge(nn ...expr.Iterator) (out expr.TypedValue, err error) {
+	rv := EmptyComposeRecordValues()
+
+	nn = append([]expr.Iterator{t}, nn...)
+
+	for _, i := range nn {
+		err = i.Each(func(k string, v expr.TypedValue) error {
+			var (
+				rVal types.RecordValue
+				bb   []byte
+			)
+
+			bb, err = json.Marshal(v.Get())
+			if err != nil {
+				return err
+			}
+
+			err = json.Unmarshal(bb, &rVal)
+			if err != nil {
+				return err
+			}
```

Comment:
I'm not entirely sure how I feel about this, perhaps add a `@todo` if we don't have a better fix atm?